### PR TITLE
Fix pytests and make TS conversion conditional

### PR DESCRIPTION
**Pull request by zhilingc**
Addressing issues #204 and #195.

Changes made for #204: 
- Advanced pandas required version to >=0.24.0 and added fastavro to `test-requirements.yaml`
- Removed test `tests/sdk/utils/test_bq_utils.py::test_query_to_dataframe` because it interfaces with BQ and is therefore not a unit test.
- Changed the python sdk to use [pandas nullable int types](https://pandas.pydata.org/pandas-docs/stable/user_guide/integer_na.html) instead of numpy int types
- BQ get table util now does not include granularity in the table name
- Changed importer expected type from `file` to `file.csv`, removed `format` from options

Changes made for #195:
- Added conditional checking if the timestamp column is declared, if not, will not run timestamp conversion
- Added test for that case

### CODE REVIEW

File: sdk/python/feast/sdk/utils/bq_util.py [hunk 1]
```diff
@@ -74,7 +74,8 @@ def get_table_name(feature_id, storage_spec):
     except KeyError:
         raise ValueError("storage spec has empty project or dataset option")
 
-    table_name = "_".join(feature_id.split(".")[:2])
+    print(feature_id)
```
Comment (davidheryanto) [inline sdk/python/feast/sdk/utils/bq_util.py]: I think the `print` is not necessary here.
I guess this was for debugging last time?

Comment (zhilingc) [reply sdk/python/feast/sdk/utils/bq_util.py]: Nice catch

**Pull request merged**